Source/Core/Core/PowerPC/Interpreter/Interpreter_Tables.cpp

```diff
@@ -5,492 +5,492 @@
+  for (auto& tpl : s_table59)
   {
-    int op = tpl.opcode;
-    m_op_table59[op] = tpl.Inst;
-    m_infoTable59[op] = &tpl.opinfo;
-  }
-
-  for (auto& tpl : table63)
+    ASSERT(table[tpl.opcode] == Interpreter::unknown_instruction);
+    table[tpl.opcode] = tpl.fn;
+  };
+  return table;
+}
+();
+constexpr std::array<Interpreter::Instruction, 1024> s_interpreter_op_table63 = []() consteval
+{
+  std::array<Interpreter::Instruction, 1024> table{};
+  Common::Fill(table, Interpreter::unknown_instruction);
+  for (auto& tpl : s_table63)
   {
-    int op = tpl.opcode;
-    m_op_table63[op] = tpl.Inst;
-    m_infoTable63[op] = &tpl.opinfo;
-  }
+    ASSERT(table[tpl.opcode] == Interpreter::unknown_instruction);
+    table[tpl.opcode] = tpl.fn;
+  };
 
-  for (int i = 0; i < 32; i++)
+  for (u32 i = 0; i < 32; i++)
   {
-    int fill = i << 5;
-    for (auto& tpl : table63_2)
+    const u32 fill = i << 5;
+    for (const auto& tpl : s_table63_2)
     {
-      int op = fill + tpl.opcode;
-      m_op_table63[op] = tpl.Inst;
-      m_infoTable63[op] = &tpl.opinfo;
+      const u32 op = fill + tpl.opcode;
+      ASSERT(table[op] == Interpreter::unknown_instruction);
+      table[op] = tpl.fn;
     }
   }
 
-  m_numInstructions = 0;
-  for (auto& tpl : primarytable)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table4_2)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table4_3)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table4)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table31)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table19)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table59)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table63)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table63_2)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-
-  initialized = true;
+  return table;
+}
+();
+
+Interpreter::Instruction Interpreter::GetInterpreterOp(UGeckoInstruction inst)
+{
+  // Check for the appropriate subtable ahead of time.
+  // (This is used by the cached interpreter and JIT, and called once per instruction, so spending a
+  // bit of extra time to optimise is worthwhile)
+  Interpreter::Instruction result = s_interpreter_op_table[inst.OPCD];
+  if (result == Interpreter::RunTable4)
+    return s_interpreter_op_table4[inst.SUBOP10];
+  else if (result == Interpreter::RunTable19)
+    return s_interpreter_op_table19[inst.SUBOP10];
+  else if (result == Interpreter::RunTable31)
+    return s_interpreter_op_table31[inst.SUBOP10];
+  else if (result == Interpreter::RunTable59)
+    return s_interpreter_op_table59[inst.SUBOP5];
+  else if (result == Interpreter::RunTable63)
+    return s_interpreter_op_table63[inst.SUBOP10];
+  else
+    return result;
+}
```

Comment:
... and a second, much more insidious thing also happens: because `RunTable4`, `RunTable19`, `RunTable31`, and `RunTable63` all had the same body (`RunTable59` used `SUBOP5` so it was still different), the compiler is allowed to treat them as the same function, and then they end up having the same address, so *this* also ended up doing the wrong thing. However, the compiler is not *required* to treat them as the same function, so this only happened on release builds, not debug builds.

On x86-64, `GetInterpreterOp` was getting called from a `7c1083a6` `mtspr IBAT0U, r0` instruction (and the JIT falls back to the interpreter for all but a few common `mtspr` instructions). So that's how this was getting called.

The instruction you had is a `blr`, which doesn't have the same fallback behavior, but because we don't use the JIT while running the apploader (compare #10923) the same previously mentioned issue with the RunTable functions applied (`RunInterpreterOp` would be used in that path, not `GetInterpreterOp`).